docs/contributing/backward-compatibility-promise.md

```diff
@@ -0,0 +1,79 @@
+# Backward Compatibility Promise
+
+Smooth and safe upgrades of your own e-commerce project are very important to us.
+In the same time, we need to be able to improve Shopsys Framework for you by adding functionality, enhancing or simplifying current functions and fixing bugs.
+After reading this promise you'll understand backward compatibility, what changes you can expect and how we plan to make changes in the future.
+
+*Note: This BC promise becomes effective with the first stable release (`7.0.0`).*
+
+## Releases and Versioning
+This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html), which means it's release versions are in format `MAJOR.MINOR.PATCH`:
+
+- `MAJOR` version may contain incompatible changes
+- `MINOR` version may add new functionality in a backward-compatible manner
+- `PATCH` version contains only backward-compatible bug fixes
+
+Even though we will try to follow these rules the best we can, it could happen that a BC breaking change is introduced in a `MINOR` or `PATCH` release.
```

Comment:
```suggestion
Even though we follow these rules, it could rarely happen that a BC breaking change is introduced in a `MINOR` or `PATCH` release.
```

I suggest to emphasize that we are not trying, we are doing! ;-)

Reply:
fair point, reworded as:

```diff
- Even though we will try to follow these rules the best we can, it could happen ...
+ Even though we keep these rules as best we can, it could happen ...
```